src/services/rename.ts

```diff
@@ -60,6 +71,37 @@ namespace ts.Rename {
         return program.isSourceFileDefaultLibrary(sourceFile) && fileExtensionIs(sourceFile.fileName, Extension.Dts);
     }
 
+    function wouldRenameAcrossExternalLibrary(originalFile: SourceFile, symbol: Symbol, checker: TypeChecker, preferences: UserPreferences): boolean {
+        if (!preferences.providePrefixAndSuffixTextForRename && symbol.flags & SymbolFlags.Alias) {
+            const importSpecifier = find(symbol.declarations!, decl => isImportSpecifier(decl));
+            if (importSpecifier && !(importSpecifier as ImportSpecifier).propertyName) {
+                symbol = checker.getAliasedSymbol(symbol);
+            }
+        }
+        const { declarations } = symbol;
+        const originalPackage = getPackagePath(originalFile.path);
+        if (originalPackage === undefined) { // original source file is not in node_modules
+            return some(declarations, declaration => !!getPackagePath(declaration.getSourceFile().path));
+        }
+        // original source file is in node_modules
+        return some(declarations, declaration => {
+            const declPackage = getPackagePath(declaration.getSourceFile().path);
+            if (declPackage) {
+                return comparePaths(originalPackage, declPackage) !== Comparison.EqualTo;
+            }
+            return false;
+        });
+    }
+
+    function getPackagePath(filePath: Path): string | undefined {

```

Comment:
Nit: If string you search for `"node_modules"`, I think you'll find related helpers, one or more of which may already do this.

Nit: Converting strings to path components and back frequently shows up in profiles.  One way to make this cleaner might be to leave both of the paths being compared as components - I'm pretty sure that's what `comparePaths` is doing anyway.